src/hypothesis/strategies.py

```diff
@@ -192,8 +194,8 @@ def floats(min_value=None, max_value=None):
             BoundedFloatStrategy() |
             ExponentialFloatStrategy() |
             JustIntFloats() |
-            NastyFloats() |
-            FullRangeFloats()
+            NastyFloats(allow_nan, allow_infinity) |
+            FullRangeFloats(allow_nan)
```

Comment:
I think this will do the wrong thing with very low probability when allow_nan=True but allow_infinity=False?